sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs

```diff
@@ -69,9 +69,9 @@ protected MetricsClient()
         /// <param name="options">The <see cref="MetricsQueryResourcesOptions"/> to configure the query.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
         /// <returns>A time series metrics result for the requested metric names.</returns>
-        public virtual Response<MetricsQueryResourcesResult> QueryResources(IEnumerable<ResourceIdentifier> resourceIds, List<string> metricNames, string metricNamespace, MetricsQueryResourcesOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response<MetricsQueryResourcesResult> QueryResources(IEnumerable<ResourceIdentifier> resourceIds, IEnumerable<string> metricNames, string metricNamespace, MetricsQueryResourcesOptions options = null, CancellationToken cancellationToken = default)
         {
-            if (resourceIds.Count() == 0 || metricNames.Count == 0)
+            if (resourceIds.Count() == 0 || metricNames.Count() == 0)
```

Comment:
Nit - should add an `Argument.IsNotNull` for these before calling a method on them

Reply:
the lists cannot be null because they are instantiated but I do check if the count is 0

Reply:
A customer could always do something like:
```csharp
IEnumerable<ResourceIdentifier> ids = null;
if (ReturnsTrueMostOfTheTime())
{
    ids = GetIds();
}
client.QueryResources(ids, ["foo"], "fooNs");
```

It's not critical to fix this release that we want out ASAP, but something we'd ideally fix in the future.  (And it's not a breaking change to go from a NullRefException to an ArgumentNullException.)**